toolchain/sem_ir/typed_insts.h

```diff
@@ -680,6 +680,16 @@ struct ExportDecl {
   InstId value_id;
 };
 
+// Allows pointing to a source location in C++ in diagnostics.
+struct ClangDiagnostic {
+  static constexpr auto Kind = InstKind::ClangDiagnostic.Define<Parse::NodeId>(
+      {.ir_name = "clang_diag",
+       .constant_kind = InstConstantKind::Never,
+       .is_lowered = false});
+
+  ClangSourceLocationId clang_loc_id;
+};
```

Comment:
I think we probably don't want distinct instructions for these. Instead, we should probably be storing locations in their own array.

Reply:
Done.